### PR TITLE
Adding framework support for Xcode project

### PR DESCRIPTION
**Pull request by matt-vanderkolk**
- Added XAsyncAll.h umbrella header to support framework header redirection.

### CODE REVIEW

File: Include/httpClient/XAsyncAll.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+#include <httpClient/config.h>
+#include <httpClient/pal.h>
+
+#if (!defined(HC_LINK_STATIC) || HC_LINK_STATIC == 0) && HC_PLATFORM == HC_PLATFORM_IOS
+#include <httpClient/XAsync.h>
+#include <httpClient/XAsyncProvider.h>
+#include <httpClient/XTaskQueue.h>
+#else
+#include <XAsync.h>
```
[the remark this answers is not in the document]
Comment (jasonsandlin) [reply Include/httpClient/XAsyncAll.h]: Perhaps async.h? 

---
In reply to: [257091313](https://github.com/Microsoft/libHttpClient/pull/375#discussion_r257091313) [](ancestors = 257091313)

File: Build/libHttpClient.Apple.C/iOSFramework/exports.txt [hunk 2]
```diff
@@ -0,0 +1,142 @@
+#
```
Comment (DrTacoMD) [inline Build/libHttpClient.Apple.C/iOSFramework/exports.txt]: exports.txt -> export.exp

File: Include/httpClient/XAsyncAll.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (matt-vanderkolk) [reply Include/httpClient/XAsyncAll.h]: async.h it is

File: Build/libHttpClient.Apple.C/iOSFramework/exports.txt (hunk 2 above)
Comment (matt-vanderkolk) [reply Build/libHttpClient.Apple.C/iOSFramework/exports.txt]: done

**Pull request merged**